### PR TITLE
GRF-69 category microfrontend

### PR DESCRIPTION
**Pull request by antoinetrouve**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

- [x] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: src/Akeneo/Category/front/src/feature/helpers/permissionsHelper.test.ts [hunk 1]
```diff
@@ -64,7 +61,7 @@ const formData: EditCategoryForm = {
   },
   errors: [],
   _token: {
-    label: 'token',
+    //label: 'token',
```
Comment (olivierpontier) [inline src/Akeneo/Category/front/src/feature/helpers/permissionsHelper.test.ts:64]: Does commenting on this line help to run the test?

File: src/Akeneo/Category/front/src/FakePIM.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+const Page = styled.div`
+  flex: 1;
+  padding: 40px;
+`;
+
+const FakePIM = ({children}) => {
+  const deps = useDependenciesContext();
+
+  useEffect(() => {
+    deps.router = {
```
Comment (olivierpontier) [inline src/Akeneo/Category/front/src/FakePIM.tsx:33]: Is the mutation of the router intended to adapt the navigation inside the microfrontend?

**Comment by olivierpontier** [comment]
We can see in the diff a blank line is missing at the end of several files Could you check your IDE configuration to homogenize this practice, please? In PHPStorm for example, you can activate the "Ensure every saved file ends with a line break"  option on Save (see in Editor > General > On Save).

File: src/Akeneo/Category/front/src/feature/helpers/permissionsHelper.test.ts (hunk 1 above)
Comment (FSAkeneo) [reply src/Akeneo/Category/front/src/feature/helpers/permissionsHelper.test.ts:64]: :+1:  no, we can remove the line
the EditCategoryForm type has no label field. The type checking is less restrictive on the type and allow unknown fields.

File: src/Akeneo/Category/front/src/FakePIM.tsx (hunk 2 above)
Comment (FSAkeneo) [reply src/Akeneo/Category/front/src/FakePIM.tsx:33]: yes these methods are to be implemented by the injected router on microfrontend.

**Comment by FSAkeneo** [comment]
> We can see in the diff a blank line is missing at the end of several files. Could you check your IDE configuration to homogenize this practice, please? In PHPStorm for example, you can activate the "Ensure every saved file ends with a line break" option on Save (see in Editor > General > On Save).

thanks, I have set the pref at user scope in vscode, it should not bother us again

**Review by olivierpontier — APPROVED** [review]

**Review by LaurentPetard — APPROVED** [review]

**Pull request merged**